docs/book/src/developer/repository-layout.md

```diff
@@ -1,3 +1,175 @@
+│   
+└───hack
+│   
+└───internal
+│   
+└───scripts
+│
+└───test
+│   │   e2e
+│   │   framework
+│   └───infrastructure
+│       │   container
+│       │   docker
+│   
+└───util
+│   
+└───version
+│   
+└───webhooks
+│   
+└───main.go
+│   
+└───Makefile
+│
+└───tilt-settings.json
+```
+
+### API
+
+[~/api](https://github.com/kubernetes-sigs/cluster-api/tree/main/api)
+
+This folder is used to store types and their related resources present in CAPI core. It includes things like API types, spec/status definitions, condition types, simple webhook implementation, autogenerated, deepcopy and conversion files. Some examples of Cluster API types defined in this package include Cluster, ClusterClass, Machine, MachineSet, MachineDeployment and MachineHealthCheck.
+
+API folder has subfolders like v1alpha3, v1alpha4 and v1beta1. These packages contain API schema definitions for the respective API groups.
+
+### Bootstrap
+
+[~/bootstrap](https://github.com/kubernetes-sigs/cluster-api/tree/main/bootstrap)
+
+This folder contains  Cluster API bootstrap provider Kubeadm (CABPK) which is another kubebuilder project. It is responsible for generating a cloud-init script to turn a Machine into a Kubernetes Node. Though it is a separate provider, Cluster API contains replica of CABPK within itself to simplify the development, testing and release process. Archived repository of CABPK can be found [here](https://github.com/kubernetes-retired/cluster-api-bootstrap-provider-kubeadm).
+
+### ControlPlane
+
+[~/controlplane](https://github.com/kubernetes-sigs/cluster-api/tree/main/controlplane)
+
+Similar to bootstrap folder, this folder also contains a reference implementation of another kubebuilder project used by CAPI i.e. KubeadmControlPlane controller. This Kubeadm provider is used to instantiate a Kubernetes control plane. 
+
+### Cluster API Provider Docker 
+
+[~/test/insfrastructure/docker](https://github.com/kubernetes-sigs/cluster-api/tree/main/test/infrastructure/docker)
+
+This folder contains a kubebuilder project i.e. CAPD which is a reference implementation of an infrastructure provider for the Cluster API project using Docker. This provider is intended for development purposes only. 
+
+Cluster API has all the three components needed to run a Cluster API management cluster - 
+CABPK (bootstrap provider), KCP (KubeAdm Control Plane provider) and CAPD (Infrastructure provider using Docker). Hence, Cluster API project has all batteries included. All these components are released together along with CAPI releases. This simplifies the release process significantly as we do not have to manage and coordinate releases for each of these providers separately.
+
+### Clusterctl CLI
+
+[~/cmd/clusterctl](https://github.com/kubernetes-sigs/cluster-api/tree/main/cmd/clusterctl)
+
+This folder contains Clusterctl, a CLI which is used to simplify the bootstrapping of a  Cluster API management cluster by providing easy interface to the users.
+
+### Configuration
```

Comment:
What about taking about "manifest generation" instead of "configuration" which is sort of too generic